src/data_gradients/feature_extractors/common/image_duplicates.py

```diff
@@ -0,0 +1,264 @@
+
+    def _is_valid_dup(self, dup_clique: List[str]) -> bool:
+        """
+        Whether the dup_clique contains duplicate image paths from valid_image_dir.
+
+        :param dup_clique: str, list of duplicated image paths
+        """
+        return len([d for d in dup_clique if d.startswith(self.valid_image_dir)]) > 1
+
+    def _is_intersection_dup(self, dup_clique: List[str]) -> bool:
+        """
+        Whether the dup_clique contains duplicate image paths from train_image_dir and valid_image_dir.
+
+        :param dup_clique: str, list of duplicated image paths
+        """
+        return len([d for d in dup_clique if d.startswith(self.train_image_dir)]) > 0 and len([d for d in dup_clique if d.startswith(self.valid_image_dir)]) > 0
+
+    @staticmethod
+    def _count_dup_appearences(dups: List[List[str]]) -> int:
+        """
+        Counts the duplicate appearences = sum of the sizes of all duplicate cliques in dups.
+        """
+        return sum([len(d) for d in dups])
+
+    def _count_dir_dup_appearences(self, dups: List[List[str]], dir: str) -> int:
+        """
+        Counts the duplicate appearences = sum of the sizes of all duplicate cliques in dups.
+        """
+        return self._count_dup_appearences(list(map(lambda dup: [d for d in dup if d.startswith(dir)], dups)))
+
+    def aggregate(self) -> Feature:
+        self._find_duplicates()
+        feature = Feature(
+            data=None,
+            plot_options=None,
+            json={"Train duplicates": self.train_dups, "Validation duplicates": self.valid_dups, "Intersection duplicates": self.intersection_dups},
+        )
+        return feature
+
+    @property
+    def title(self) -> str:
+        return "Image Duplicates"
+
+    @property
+    def description(self) -> str:
+        desc = self._get_split_description(self.train_dups, "Train", self.train_dups_appearences)
+        if self.valid_image_dir is not None:
+            desc += self._get_split_description(self.valid_dups, "Validation", self.validation_dups_appearences)
+            desc += f"<br /><br />There are {len(self.intersection_dups)} duplicates between train and validation."
+            if len(self.intersection_dups):
+                desc = desc.replace(
+                    "train and validation.",
+                    f"train and validation appearing {self.intersection_train_appearnces} times in the train image directory,"
+                    f" and {self.intersection_val_appearnces} times in the validation image directory.",
+                )
+
+        else:
+            desc += "<br />"
+        return desc
+
+    def _get_split_description(self, dups: List, split: str, appearences: int) -> str:
+        desc = f"<strong>{split} duplicated images</strong>:<br /> There are {len(dups)} duplicated images.<br />"
+        if len(dups) > 0:
+            desc = desc.replace(".<br />", f" appearing {appearences} times across the dataset.<br /><br />")
```

Comment:
probably a cleaner more scalable way to use `<br>` is to do `.replace("\n", "<br />")` in the AbstractManager

Reply:
Done in latest commit.